recipe/meta.yaml

```diff
@@ -11,21 +11,26 @@ source:
 
 build:
   noarch: python
-  number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  number: 1
+  script: |
+    set -euxo pipefail
+    tree -a ./
+    rm -f yarn.lock
```

Comment:
Is there an issue with the current `yarn.lock` in the source repo?

Reply:
Not sure - don't know much about JS packaging. I figured if I nuked it, it might recreate it in the CI.

> *YN0028: The lockfile would have been created by this install, which is explicitly forbidden.*

Reply:
It doesn't look like there is a `yarn.lock` file at all?
```
+ tree -a ./
./
├── build_env_setup.sh
├── conda_build.sh
├── install.json
├── LICENSE
├── MANIFEST.in
├── metadata_conda_debug.yaml
├── package.json
├── PKG-INFO
├── pyproject.toml
├── README.md
├── setup.cfg
├── setup.py
├── src
│   └── index.ts
├── style
│   ├── codemirror-darcula.css
│   ├── index.css
│   ├── scrollbars.css
│   └── variables.css
├── theme-darcula
│   ├── __init__.py
│   ├── labextension
│   │   ├── package.json
│   │   ├── static
│   │   │   ├── 568.d91f2ec64631eec2f047.js
│   │   │   ├── remoteEntry.8a4c97071917cf84832f.js
│   │   │   ├── style.js
│   │   │   └── third-party-licenses.json
│   │   └── themes
│   │       └── @telamonian
│   │           └── theme-darcula
│   │               ├── index.css
│   │               └── index.js
│   └── _version.py
├── theme_darcula.egg-info
│   ├── dependency_links.txt
│   ├── not-zip-safe
│   ├── PKG-INFO
│   ├── requires.txt
│   ├── SOURCES.txt
│   └── top_level.txt
└── tsconfig.json
```

Maybe that's the problem?

Reply:
Also probably the `pip install .` command uses the yarn provided by JupyterLab via `jlpm`.

In JupyterLab 4 it was updated to Yarn 3. But the source `yarn.lock` is still Yarn 1: https://github.com/telamonian/theme-darcula/blob/master/yarn.lock

Reply:
> Also probably the `pip install .` command uses the yarn provided by JupyterLab via `jlpm`.

Looks like it does:

![image](https://github.com/conda-forge/theme-darcula-feedstock/assets/591645/2e15d910-a7cc-4de2-ae07-30eac7429d7a)

Reply:
Maybe we can try to add `yarn=1` to the `build` requirements and see if it gets picked up by `jupyter-packaging`.

Reply:
Do you know how to ~~hack~~ fix it?

Reply:
Ideally the upstream repo would be updated to use Yarn 3. Also also `hatch` since `jupyter-packaging` is deprecated.

So maybe worth opening a PR there. There is the migration script which can help update the repo: `python -m jupyterlab.upgrade_extension .`

Reply:
Otherwise adding `yarn=1` to the `build` here and trying to alias `jlpm=yarn` could work maybe?

Reply:
Hmm, I might try updating upstream when I can find the time...